docs/building-extensions/libraries/basic-library.md

```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 2
+title: Create a Joomla Library
+---
+
+Creating a Joomla Library
+======================================
+
+A Joomla library is a reusable set of classes or functions that can be shared across multiple extensions. Libraries help in organizing and reusing code, making development more efficient and maintainable.
+
+In this guide, we will create a Joomla library and learn how to use it in Joomla projects.
+
+## Folder Structure
+
+The folder structure for a Joomla library is as follows:
+
+```
+libraries/
+    yourlibraryname/
+        src/
+            YourLibrary.php
+        yourlibraryname.xml
+```
+
+### Key Sections of the Library
+
+- `src/YourLibrary.php`: This is the main file of the library. It contains the class definition and the functions that the library will provide.
+- `yourlibraryname.xml`: This file contains the metadata for the library.
+
+### Extension manifest file
+
+```xml title="libraries/yourlibraryname/yourlibraryname.xml"
+<?xml version="1.0" encoding="utf-8"?>
+<extension type="library" method="upgrade">
+    <name>Joomla 5 Example Library</name>
+    <libraryname>JoomlaExampleLibrary</libraryname>
+    <author>Author Name</author>
+    <creationDate>2024-09-08</creationDate>
+    <license>GNU General Public License version 2 or later</license>
+    <version>1.0.0</version>
+    <description>Showcase on how to create a Joomla 5 library to reuse code.</description>
+    <namespace path="src">VendorName\LibraryName</namespace>
+    <files>
+        <folder>src</folder>
+        <filename>joomlaexamplelibrary.xml</filename>
```

Comment:
You need to remove this line

Reply:
which one, the namespace?
@Fedik suggested i should add it :D I'm confused now

Reply:
ok, will remove it

Reply:
No, the `<filename>` line, as that file doesn't exist.